docs/architecture/devops-for-aspnet-developers/actions-build.md

```diff
@@ -0,0 +1,193 @@
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v1
+      with:
+        dotnet-version: 5.0.x
+    - name: Restore dependencies
+      run: dotnet restore
+    - name: Build
+      run: dotnet build --no-restore
+    - name: Test
+      run: dotnet test --no-build --verbosity normal
+```
+
+You can see the following:
+
+1. There is a `name` that names the workflow.
+1. The `on` object specifies when this workflow should run. This workflow has two events that trigger it: `push` to `main` and `pull_request` to `main`. Each time someone commits to `main` or creates a pull request (PR) to `main`, this workflow will execute.
+1. There is a single `job` called `build`. This build should run on a hosted agent. `ubuntu_latest` specifies the most recent Ubuntu hosted agent.
+1. There are five steps:
+    1. `acions/checkout@2` is an action that checks out the code in the repo onto the runner.
+    1. `actions/setup-dotnet@v1` is an action that sets up the .NET CLI. This step also specifies a `name` attribute for the logs and the `dotnet-version` parameter within the `with` object.
+    1. Three `run` steps that execute `dotnet restore`, `dotnet build`, and `dotnet test`. `name` attributes are also specified for these `run` steps to make the logs look pretty.
+
+## Publish the output
+
+Now that you've successfully built and tested the code, add steps that publish the output so you can deploy the web app.
+
+1. Navigate to the `.github/workflows/dotnet.yml` file and select the pencil icon to edit it.
+
+    ![Edit the YML file](./media/actions/build/click-edit.jpg)
+    **Figure 5** Edit the YML file.
+
+1. Add the following step at the bottom of the file, below the `Test` step, to run the `dotnet publish` command to publish the website:
+
+  ```yml
+      - name: Test
+        run: dotnet test --no-build --verbosity normal # <-- this is the current bottom line
+
+      - name: Publish
+        run: dotnet publish SimpleFeedReader/SimpleFeedReader.csproj -c Release -o website
+  ```
+
+1. This publishes the web app to a folder on the hosted agent. We now want to _upload_ the site as a build artifact that we can deploy to Azure. To do this activity, we are going to use an existing action.
+1. On the list of Actions in the Actions Helper pane on the right, search for `artifact` and click on the `Upload a Build Artifact (By actions)` action.
+
+    ![Accessing the Actions helper](./media/actions/build/search-upload-artifact.jpg)
+    **Figure 6** Accessing the snippet helper.
+
+1. Edit the version to `v2.2.2` to display a sample snippet. Click the clipboard icon to copy the snippet and paste it into the workflow below the publish step.
+
+    ![Copying a snippet](./media/actions/build/copy-snippet.jpg)
+    **Figure 8** Copying a snippet.
+
+1. Edit the YAML for this step to look as follows:
+
+  ```yml
```

Comment:
Tab-indent this entire code block, including the code fences. This will fix the visual separation from the line item that I currently see:

![image](https://user-images.githubusercontent.com/10702007/110996357-70972a80-8341-11eb-9c50-2e869ce4c8d2.png)